packages/driver/src/cy/commands/screenshot.ts

```diff
@@ -338,8 +332,10 @@ const takeScreenshot = (Cypress, state, screenshotConfig, options: TakeScreensho
 
       return null
     })
-    .then(() => {
-      return sendAsync('before:screenshot', getOptions(true))
+    .then(async () => {
+      // we need to await this so that JS running to prepare for the screenshot
+      // is fully completed before we start measuring the viewport etc
+      await send('before:screenshot', getOptions(true), () => {})
```

Comment:
Do we need this? @marktnoonan.

Reply:
Let's look into it - seems like it's not needed, I'll try removing it and see what happens.

Reply:
Agreed, it looks like other parts of this branch have made this change no longer needed.